### PR TITLE
Various tests, utils, and refactorings

### PR DESCRIPTION
**Pull request by mbland**
If this is too much for one PR, happy to split out orthogonal pieces. All of this is refactoring and testing with an eye towards moving more functionality into the [team_hub gem](https://github.com/18F/team_hub).

@afeld I know you're going on vacation, so no rush.
@adelevie In case you have cycles, love to see what you think of some of my tricks, particularly regarding `TEST_TMPDIR` and  `_test/temp_file_helper.rb`.
@gboone In case you'd like to take a look.

### CODE REVIEW

File: _data/import-public-snippets.rb [hunk 1]
```diff
@@ -25,16 +25,19 @@
...
 
-site = ::Jekyll::Site.new ::Jekyll::Configuration::DEFAULTS
-site.read_data('.')
-joiner_impl = ::Hub::JoinerImpl.new site
+team_filename = File.join('private', 'team.yml')
+team = SafeYAML.load_file(team_filename, :safe=>true)
+unless team
+  puts "Failed to parse #{team_fileanme}"
+  exit 1
+end
```
Comment (afeld) [inline _data/import-public-snippets.rb]: Minor: maybe put all of this in a `get_team` method or something?

File: _test/private_assets_copy_team_images_test.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# <https://creativecommons.org/publicdomain/zero/1.0/>.
+#
+# @author Mike Bland (michael.bland@gsa.gov)
+
+require_relative "test_helper"
+require_relative "../_plugins/private_assets"
+require_relative "site"
+require_relative "temp_file_helper"
+
+require "minitest/autorun"
```
Comment (afeld) [inline _test/private_assets_copy_team_images_test.rb]: If a bunch of these requires are common to many test files, might as well move them into the `test_helper`.

**Comment by afeld** [comment]
"Vacation" is that thing where you work from somewhere other than you live, right? Or is that just me? :snowboarder: :snowflake: :wink: 

Just a couple minor notes, but otherwise :shipit:

File: _data/import-public-snippets.rb (hunk 1 above)
Comment (mbland) [inline _data/import-public-snippets.rb]: Done; called it `load_team_data`.

File: _test/private_assets_copy_team_images_test.rb (hunk 2 above)
Comment (mbland) [inline _test/private_assets_copy_team_images_test.rb]: I'm big on explicit, narrowly-scoped `require` statements in each file, to reduce dependencies and the chance that removing a dependency in one file will break another. Maybe that's paranoid on my part, but it helps me sleep at night! :-)

**Pull request merged**